sound/soc/intel/boards/sof_da7219.c

```diff
@@ -232,34 +226,23 @@ static int hdmi_init(struct snd_soc_pcm_runtime *rtd)
 {
 	struct card_private *ctx = snd_soc_card_get_drvdata(rtd->card);
 	struct snd_soc_dai *dai = asoc_rtd_to_codec(rtd, 0);
-	struct hdmi_pcm *pcm;
-
-	pcm = devm_kzalloc(rtd->card->dev, sizeof(*pcm), GFP_KERNEL);
-	if (!pcm)
-		return -ENOMEM;
-
-	pcm->device = dai->id;
-	pcm->codec_dai = dai;
 
-	list_add_tail(&pcm->head, &ctx->hdmi_pcm_list);
+	ctx->hdmi.hdmi_comp = dai->component;
 
 	return 0;
 }
 
 static int card_late_probe(struct snd_soc_card *card)
 {
 	struct card_private *ctx = snd_soc_card_get_drvdata(card);
-	struct snd_soc_acpi_mach *mach = (card->dev)->platform_data;
-	struct hdmi_pcm *pcm;
-
-	if (mach->mach_params.common_hdmi_codec_drv) {
-		pcm = list_first_entry(&ctx->hdmi_pcm_list, struct hdmi_pcm,
-				       head);
-		return hda_dsp_hdmi_build_controls(card,
-						   pcm->codec_dai->component);
-	}
 
-	return -EINVAL;
+	if (!ctx->hdmi.idisp_codec)
+		return 0;
+
+	if (!ctx->hdmi.hdmi_comp)
+		return -EINVAL;
+
+	return hda_dsp_hdmi_build_controls(card, ctx->hdmi.hdmi_comp);
```

Comment:
same here, can we not remove the test on the common codec driver and then add the new structures?

Reply:
You mean keeping the test of mach->mach_params.common_hdmi_codec_drv? I though it is always true for upstream kernel?

Reply:
> I though it is always true for upstream kernel?

I am not sure about it. I can see [mach_params->common_hdmi_codec_drv = true;](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda.c#L1481). It seems common_hdmi_codec_drv is always true in SOF, but I don't know what if `CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC` is not enabled. @kv2019i Can you comment on it?

Reply:
> You mean keeping the test of mach->mach_params.common_hdmi_codec_drv? I though it is always true for upstream kernel?

I meant doing exactly what you are now suggesting so I am happy with the code.